cmd/skopeo/login.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"io"
+	"os"
+
+	"github.com/containers/common/pkg/auth"
+	"github.com/containers/image/v5/types"
+	"github.com/spf13/cobra"
+)
+
+type loginOptions struct {
+	global    *globalOptions
+	loginOpts auth.LoginOptions
+	getLogin  optionalBool
+	tlsVerify optionalBool
+}
+
+func loginCmd(global *globalOptions) *cobra.Command {
+	opts := loginOptions{
+		global: global,
+		loginOpts: auth.LoginOptions{
+			Stdin:  os.Stdin,
+			Stdout: os.Stdout,
+		},
+	}
+	cmd := &cobra.Command{
+		Use:     "login",
+		Short:   "Login to a container registry",
+		Long:    "Login to a container registry on a specified server.",
+		RunE:    commandAction(opts.run),
+		Example: `skopeo login quay.io`,
+	}
+	adjustUsage(cmd)
+	flags := cmd.Flags()
+	optionalBoolFlag(flags, &opts.getLogin, "get-login", "Return the current login user for the registry")
+	optionalBoolFlag(flags, &opts.tlsVerify, "tls-verify", "require HTTPS and verify certificates when accessing the registry")
+	flags.AddFlagSet(auth.GetLoginFlags(&opts.loginOpts))
+	return cmd
+}
+
+func (opts *loginOptions) run(args []string, stdout io.Writer) error {
+	ctx, cancel := opts.global.commandTimeoutContext()
+	defer cancel()
+	sys := opts.global.newSystemContext()
+	sys.DockerCertPath = opts.loginOpts.CertDir
+	sys.AuthFilePath = opts.loginOpts.AuthFile
+	if opts.tlsVerify.present {
+		sys.DockerDaemonInsecureSkipTLSVerify = !opts.tlsVerify.value
+		sys.DockerInsecureSkipTLSVerify = types.NewOptionalBool(!opts.tlsVerify.value)
+	}
+	opts.loginOpts.GetLoginSet = opts.getLogin.present
```

Comment:
This sets `GetLoginSet` to `true` if the user uses `--get-login=false`.

Was there a reason to treat `--get-login` and `--password-stdin` differently?

Reply:
Document doesn't describe --get-login as a bool value that can be true or false. what it should do if --get-login=false?
```
# podman login --get-login docker.io
username
# podman login --get-login=false docker.io
username
# podman login --get-login=true docker.io
username
```

Reply:
That’s a very good point; unfortunately, unlike C’s `getopt` and the like, neither https://golang.org/pkg/flag/ nor https://godoc.org/github.com/spf13/pflag seems to actually implement parameter-less options; the best we have is `NoOptDefVal`.

I think we can either interpret `--get-login=false` as if it were missing (`GetLoginSet = opts.getLogin.value`) or outright reject it (`if opts.getLogin.present && !opts.getLogin.value { fail }`); treating `--get-login=false` as if the user did want to do `--get-login` seems extremely unintuitive to me.

I don’t see any way at all, given the current API, to define a flag that can be used without a value but that rejects any use with a value (we can se `NoOptDefVal` to anything, and reject uses with any other value, but the user can always call the binary with the same `NoOptDefVal` value explicitly).

Overall, given the way things are in Go, and the way _everything else_, AFAIK, just uses a `pflag.Bool*` and checks the value for `true`, I don’t think spending extra effort on this one boolean (but not on `StdinPassword`, for example), is really worthwhile.